src/Symfony/Component/Mime/Exception/UnexpectedTypeException.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mime\Exception;
+
+class UnexpectedTypeException extends InvalidArgumentException
```

Comment:
Adding a new class is a new feature, so cannot be part of a bugfix.
I'd suggest throwing a TypeError instead, as `TextPart` does already.